engine/src/main/java/org/destinationsol/game/MapDrawer.java

```diff
@@ -409,6 +423,20 @@ public void drawObjIcon(float iconSz, Vector2 position, float objAngle,
         }
     }
 
+    public void drawWaypointIcon(float iconSz, Vector2 position, TextureAtlas.AtlasRegion icon, Object drawer, Color color) {
+        float innerIconSz = iconSz * INNER_ICON_PERC;
+        //allow usage of UiDrawer or GameDrawer
```

Comment:
This is an example of a comment that describes *what* the code is doing. It's a good start, but by just reading the code I can figure that out. A good comment describes not what but *why* the code is doing what it's doing. IE why could `drawer` be more than one type of object, and why is handling them the same way (after the cast) fine?

Reply:
So should I write something like "drawer can be either UiDrawer or GameDrawer because the function is also called from BorderDrawer which has UiDrawer, not GameDrawer available"
Or just "[...] because it is called from different contexts"?

Reply:
I would lean towards something like the first option for this specific case